packages/salesforcedx-vscode-core/src/commands/templates/forceAnalyticsTemplateCreate.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  CancelResponse,
+  ContinueResponse,
+  DirFileNameSelection,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { nls } from '../../messages';
+import {
+  CompositeParametersGatherer,
+  PathStrategyFactory,
+  SelectOutputDir,
+  SfdxCommandlet,
+  SfdxWorkspaceChecker,
+  SourcePathStrategy
+} from '../util';
+import { BaseTemplateCommand } from './baseTemplateCommand';
+import {
+  ANALYTICS_TEMPLATE_DIRECTORY,
+  ANALYTICS_TEMPLATE_TYPE
+} from './metadataTypeConstants';
+
+export class ForceAnalyticsTemplateCreateExecutor extends BaseTemplateCommand {
+  constructor() {
+    super(ANALYTICS_TEMPLATE_TYPE);
+  }
+  public getFileExtension(): string {
+    return '.json';
+  }
+  public build(data: TemplateNameAndDir): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_analytics_template_create_text'))
+      .withArg('force:analytics:template:create')
+      .withFlag('--outputdir', data.outputdir)
+      .withFlag('--templatename', data.templateName)
+      .withLogName('force_analytics_template_create')
+      .build();
+  }
+
+  public sourcePathStrategy: SourcePathStrategy = PathStrategyFactory.createDefaultStrategy();
+
+  public getDefaultDirectory() {
+    return ANALYTICS_TEMPLATE_DIRECTORY;
+  }
+
+  public getPathToSource(outputDir: string, fileName: string) {
+    return super.getPathToSource(outputDir, fileName ? fileName : '');
```

Comment:
I'm not sure this is needed, can you explain this scenario where filename needs to be set to an empty string ?

Reply:
It’s because we don’t use fileName at this point . It wasn’t causing issues, but noticed error in the console.  Filene gets sent in from the framework with undefined otherwise and it types to string.  Suggest another way?

Reply:
The fileName is the name of the file that's opened once the template is generated. In this case I'm guessing it's the JSON file in the root of the bundle. I'm not sure why this method needs to be changed to public and overridden.

Reply:
no its not the same thing as the user inputs, that just ends up being the directory name.  Each template has a template-info.json file maybe I could set the fileName to this file to avoid this.